src/main/java/com/powsybl/openloadflow/network/LfContingency.java

```diff
@@ -118,6 +123,7 @@ public void apply(LoadFlowParameters parameters) {
             generator.setParticipating(false);
             if (generator.getGeneratorControlType() != LfGenerator.GeneratorControlType.OFF) {
                 generator.setGeneratorControlType(LfGenerator.GeneratorControlType.OFF);
+                participatingGeneratorsToBeRemoved.add(generator);
```

Comment:
apply method is not supposed to change the state of LfContingency but LfNetwork one. If apply is called 2 times, participatingGeneratorsToBeRemoved will be filled 2 times. Is it really something that has to be stored in LfContingency? Maybe it could be just created on the fly by filtering generators of the LfContingency?